homeassistant/components/nextcloud/__init__.py

```diff
@@ -26,10 +28,25 @@
 
 CONFIG_SCHEMA = cv.removed(DOMAIN, raise_if_present=False)
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the Nextcloud integration."""
 
+    # migrate old entity unique ids
+    entity_reg = er.async_get(hass)
+    entities: list[er.RegistryEntry] = er.async_entries_for_config_entry(
+        entity_reg, entry.entry_id
+    )
+    for entity in entities:
+        old_uid_start = f"{entry.data[CONF_URL]}#nextcloud_"
+        new_uid_start = f"{entry.data[CONF_URL]}#"
+        if entity.unique_id.startswith(old_uid_start):
+            new_uid = entity.unique_id.replace(old_uid_start, new_uid_start)
+            _LOGGER.debug("migrate unique id '%s' to '%s'", entity.unique_id, new_uid)
+            entity_reg.async_update_entity(entity.entity_id, new_unique_id=new_uid)
```

Comment:
Is the URL a valid unique id?
Should the URL not be replaced by the entry ID?

Reply:
you're right ... will prepare a PR for this later this day

Reply:
#99226